web/controllers/chatNotificationController/util.ts

```diff
@@ -0,0 +1,77 @@
+import { getChatUser } from '../../../common/chat';
+import { getLogger } from '../../../common/logger/logger';
+import { User } from '../../../common/user';
+import { getCourse, getSubcourse } from '../../../graphql/util';
+import { GroupNotificationContext, NotificationTriggered, OneOnOneNotificationContext } from './types';
+
+const logger = getLogger('ChatNotification');
+
+export enum ChatType {
+    GROUP = 'group',
+    ONE_ON_ONE = 'one_on_one',
+}
+
+export function getChatType(participants: string[]) {
+    if (participants.length === 2) {
+        return ChatType.ONE_ON_ONE;
+    }
+    return ChatType.GROUP;
+}
+
+export async function verifyChatUser(user: User) {
+    const chatUser = await getChatUser(user);
+    if (chatUser) {
+        return true;
+    }
+    return false;
+}
+
+export async function getNotificationContext(notificationBody: NotificationTriggered): Promise<GroupNotificationContext | OneOnOneNotificationContext> {
+    const { recipient, sender, conversation, messages } = notificationBody.data;
+    const firstnameRecipient: string = recipient.name.split(' ')[0];
+    const firstnameSender: string = sender.name.split(' ')[0];
+
+    const subcourseConversation = conversation.custom.subcourse ? JSON.parse(conversation.custom.subcourse) : undefined;
+    const match = conversation.custom.match ? JSON.parse(conversation.custom.match) : undefined;
+    const participants = Object.keys(conversation.participants);
+
+    const chatType = getChatType(participants);
+
+    let notificationContext: GroupNotificationContext | OneOnOneNotificationContext;
+
+    if (chatType === ChatType.ONE_ON_ONE) {
+        let courseId: number;
+        if (subcourseConversation?.length === 1) {
+            const subcourse = await getSubcourse(subcourseConversation[0]);
+            courseId = subcourse.courseId;
+        }
+
+        notificationContext = {
+            sender: { firstname: firstnameSender },
+            recipient: { firstname: firstnameRecipient },
```

Comment:
not needed, one can just use user.firstname

Reply:
the recipient is from talkjs and only has the fullname.

Reply:
But the recipient is always the user?

Reply:
yes, the notification that talkjs triggers always set the `recipient` as the `user` who missed the message. 🙂

Reply:
Yeah but we dont need it twice in the notification context, user is enough.

Reply:
I already removed `recipient` from the context